### ISSUE TITLE
Migration to Swift?

### ISSUE THREAD

**Issue by ogezue** [ealeksandrov/EAIntroView#136]
Have you planned to migrate to swift?

**Comment by ealeksandrov** [comment]
No, I can't see any pros in migrating to Swift. But I have planned to add modern obj-c features like nullability and generics to ease up the bridging with Swift projects.

**Comment by ealeksandrov** [comment]
Anyone having problems using EAIntroView with Swift - refer to answer in #143

**Comment by ealeksandrov** [comment]
Since Swift 3 is out and pretty stable - next EAIntroView major version will be rewritten in Swift.

**Comment by ealeksandrov** [comment]
Obj-c lightweight generics added in f440f6b522fc3170231cae45d49feb862fded324.

**Comment by wazowski78** [comment]
I have manage to use objective C code on a Swift project so I am attaching the code to help any other dev to solve this issue easier.

`var page2 = EAIntroPage.init()
        page2.bgColor = UIColor.white
        page2.title = "title for this page"
        page2.titleFont = UIFont.boldSystemFont(ofSize: 25)
        page2.titleColor = UIColor.black
        page2.titlePositionY = 240;
        page2.desc = "description for this page"
        page2.descFont = UIFont.systemFont(ofSize: 20)
        page2.descColor = UIColor.black
        page2.descPositionY = 240;
        page2.titleIconView = UIImageView(image: UIImage(named: "page2"))
        page2.titleIconPositionY = 120;
        
        var page3 = EAIntroPage.init()
        page3.bgColor = UIColor.white
        page3.title = "title for this page"
        page3.titleFont = UIFont.boldSystemFont(ofSize: 25)
        page3.titleColor = UIColor.black
        page3.titlePositionY = 240;
        page3.desc = "description for this page"
        page3.descFont = UIFont.systemFont(ofSize: 20)
        page3.descColor = UIColor.black
        page3.descPositionY = 240;
        page3.titleIconView = UIImageView(image: UIImage(named: "page3"))
        page3.titleIconPositionY = 120;
        
        var intro = EAIntroView(frame: self.navigationController?.view.bounds, andPages:[page2,page3])
        intro?.show(in: self.navigationController?.view, animateDuration: 0.0)
`

**Comment by Hashir524** [comment]
@wazowski78 Saved My Day
